src/line_launcher/mod.rs

```diff
@@ -222,7 +253,7 @@ impl LineLauncher {
                                 ) =>
                         {
                             playing_state = PlayingState::Playing {
-                                line_index: rand::thread_rng().gen_range(0..self.lines.len()),
+                                line_index: thread_rng.gen_range(0..self.lines.len()),
```

Comment:
Like a couple comments from the previous PR, I think this should be encapsulated in `PlayingState` somehow. In general, unless a struct/enum is really "dumb" (eg. just decoded JSON from an API), I think it should be created/manipulated by smart constructors and functions on it, rather than having its fields be directly set by other code.